R/pass_fail_condi.R

```diff
@@ -0,0 +1,65 @@
+# Fail if condition matches
+#' @export
+fail_if <- function(x, message) {
+  condi(x, message, correct = FALSE)
+}
+
+#' Pass if condition matches
+#' @export
+pass_if <- function(x, message) {
+  condi(x, message, correct = TRUE)
+}
+
+#' Condition object
+#' Captures what the user passes into \code{pass_if} or \code{fail_if},
+#' figures out what type of object was passed into \code{x},
+#' and returns an object that will be passed into \code{evaluate_condi}
+condi <- function(x, message, correct) {
+  type <-
+    if (rlang::is_formula(x)) {
+      "formula"
+    } else if (rlang::is_function(x)) {
+      "function"
+    } else {
+      "value"
+    }
+
+  ret <- list(
+    x = x,
+    message = message,
+    correct = correct,
+    type = type
+  )
+  class(ret) <- "grader_condition"
+  ret
+}
+
+#' Evaluates a condition
+#' @returns Boolean of the condition
+#' @export
+evaluate_condi <- function(condi, grader_args, learnr_args) {
+  checkmate::assert_class(condi, "grader_condition")
+  switch(condi$type,
+         "formula" = evaluate_condi_formula(condi$x, grader_args$solution_quo, learnr_args$envir_prep), # nolint
+         "function" = evaluate_condi_function(condi$x, grader_args$solution_quo),
```

Comment:
might need envir_prep. idk for sure. Not the priority right now

Reply:
not sure if this is needed since the function is called within check_result. but can come back to it if the instructor puts in a function that makes calls to variables in the environment.